File: app/src/main/java/com/yd/xzqcomponentkt/MainActivity.java
package com.yd.xzqcomponentkt;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;
import com.xzq.module_base.api.NetCallback;
import com.xzq.module_base.arouter.RouterUtils;
import com.xzq.module_base.base.BaseActivity;


public class MainActivity extends BaseActivity {

    public static void start(Context context) {
        Intent starter = new Intent(context, MainActivity.class);
        context.startActivity(starter);
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_main;
    }

    @Override
    protected void initViews(@Nullable Bundle savedInstanceState) {
        hideToolbar();
    }

    @Override
    protected void activityExit() {
    }

    public void pagingLoad(View view) {
        NetCallback.FIRST_PAGE_INDEX = 0;
        RouterUtils.openMvp();
    }

    public void pagingLoadEmpty(View view) {
        NetCallback.FIRST_PAGE_INDEX = 10000;
        RouterUtils.openMvp();
    }

    public void pagingLoadCustomLoading(View view) {
        NetCallback.FIRST_PAGE_INDEX = 335;
        RouterUtils.openMvp();
    }

    public void pagingLoadCustomEmpty(View view) {
        NetCallback.FIRST_PAGE_INDEX = 10001;
        RouterUtils.openMvp();
    }
}

File: app/src/main/java/com/yd/xzqcomponentkt/MainActivity.kt
package com.yd.xzqcomponentkt

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.View
import com.xzq.module_base.api.NetCallback
import com.xzq.module_base.arouter.RouterUtils
import com.xzq.module_base.base.BaseActivity


class MainActivity : BaseActivity() {

    override fun getLayoutId(): Int {
        return R.layout.activity_main
    }

    override fun initViews(savedInstanceState: Bundle?) {
        hideToolbar()
    }

    override fun activityExit() {}

    fun pagingLoad(view: View) {
        NetCallback.FIRST_PAGE_INDEX = 0
        RouterUtils.openMvp()
    }

    fun pagingLoadEmpty(view: View) {
        NetCallback.FIRST_PAGE_INDEX = 10000
        RouterUtils.openMvp()
    }

    fun pagingLoadCustomLoading(view: View) {
        NetCallback.FIRST_PAGE_INDEX = 335
        RouterUtils.openMvp()
    }

    fun pagingLoadCustomEmpty(view: View) {
        NetCallback.FIRST_PAGE_INDEX = 10001
        RouterUtils.openMvp()
    }

    companion object {

        fun start(context: Context) {
            val starter = Intent(context, MainActivity::class.java)
            context.startActivity(starter)
        }
    }
}
